Components/ImportService.php

```diff
@@ -399,8 +389,11 @@ public function getArticlesWithAutoImportedCategories()
         foreach ($statement->fetchAll(\PDO::FETCH_ASSOC) as $item) {
             $categories = json_decode($item['category'], true);
             if (is_array($categories) && count($categories) > 0) {
-                $articleId = $item['article_id'];
-                $remoteItems[$articleId] = [$categories, $item['shop_id']];
+                $product = new Product();
+                $product->shopId = $item['shop_id'];
+                $product->sourceId = $item['article_id'];
+                $product->categories = $categories;
+                $remoteItems[] = $product;
```

Comment:
I prefer to use ```$remoteItems[$articleId] = $product```. Then ```$product->sourceId = $item['article_id'];``` is not needed.

sourceId confusing me because it contains articleId.

Reply:
yes good point that confused me too
i wasn't happy with it either